hepcrawl/spiders/aps_spider.py

```diff
@@ -58,64 +80,289 @@ def __init__(self, url=None, from_date=None, until_date=None,
+        for editor_group in editor_groups:
+            editors.extend(editor_group.xpath(
+                './/string-name/text()').extract())
+
+        title = ref.xpath(".//article-title/text()").extract_first()
+        publication = ref.xpath(".//source/text()").extract_first()
+        fpage = ref.xpath(".//page-range/text()").extract_first()  # there's only fpage
+        issue = ref.xpath(".//issue/text()").extract_first()
+        volume = ref.xpath(".//volume/text()").extract_first()
+        year = ref.xpath(".//year/text()").extract_first()
+        urls = ref.xpath(".//ext-link/text()").extract()
+        arxiv_raw = ref.xpath(
+            './/pub-id[@pub-id-type="arxiv"]/text()').extract()
+        arxiv_id = self._format_arxiv_id(arxiv_raw)
+        publisher = ref.xpath('.//publisher-name/text()').extract_first()
+        publisher_loc = ref.xpath(
+            './/publisher-loc/text()').extract_first()
+        if not publisher_loc:
+            publisher_loc = ref.xpath('.//publisher-name/following-sibling::text()[1]').extract_first()
+        if publisher_loc:
+            publisher = publisher_loc.strip(",. ") + ': ' + publisher
+
+        issn = ref.xpath('.//issn/text()').extract_first()
+        raw_reference = ref.extract()
+
+        # Unstructured text
+        # FIXME: this is like in harvestingkit. Do we really want this?
+        # Alternative is to just use the raw_reference.
+        # This would go into reference['misc'].
+
+        # unstructured_text = []
+        # for text in ref.xpath('./text()[normalize-space()]').extract():
+        #     text = re.sub(r'[\[\]\(\.;\)]', '', text).strip()
+        #     if text.startswith(','):
+        #         text = text[1:].strip()
+        #     if text.endswith('Report No'):
+        #         if institution:
+        #             text = institution + ' ' + text
+        #             institution = ''
+        #             text = text.strip()
+        #     elif text.endswith(' ed'):
+        #         text += '.'
+        #     elif text.endswith('PhD thesis,'):
+        #         if institution:
+        #             text += ' ' + institution
+        #             institution = ''
+        #         else:
+        #             text = text[:-1]
+        #     elif text.startswith('Seminar,'):
+        #         text = institution + ' Seminar, \"' + article_title + '\"'
+        #         institution = ''
+        #     elif text == u'\u201d':
+        #         text = ''
+        #     ignore_text = ['in', 'pp', 'edited by']
+        #     if text.startswith('Vol'):
+        #         temp = re.sub(r'\D', '', text)
+        #         if temp:
+        #             volume += temp
+        #     elif len(text) > 1 and text not in ignore_text \
+        #             and not (text.isdigit() or text[:-1].isdigit()):
+        #         unstructured_text.append(text)
+        # if unstructured_text:
+        #     unstructured_text = " ".join(unstructured_text)
```

Comment:
imo the unstructured text is not needed, no, as you say the raw is enough, but I'm not 100% sure if there's any special usage for it later somewhere... @kaplun do you know?